### PR TITLE
executor: add more diagnosis rule to check some metrics exceed thresholds

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve? 
 This PR is the continuation of  PR https://github.com/pingcap/tidb/pull/14801.

This PR adds a more `threshold-check` rule, which is used to check some metrics threshold value, like `tso duration`, `get token duration`, `storage-write` duration and so on. It will detect the following metrics tables in the current implementation:

* `pd_tso_wait_duration`
* `tidb_get_token_duration`
* `tidb_load_schema_duration`
* `tikv_scheduler_command_duration`
* `tikv_handle_snapshot_duration`
* `tikv_storage_async_request_duration`
* `tikv_storage_async_request_duration`
* `tikv_engine_write_duration`
* `tikv_engine_max_get_duration`
* `tikv_engine_max_seek_duration`
* `tikv_scheduler_pending_commands`
* `tikv_block_index_cache_hit`
* `tikv_block_filter_cache_hit`
* `tikv_block_data_cache_hit`

### What is changed and how it works?

Get the metric-data and compare with the threshold.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: infoschema/metric_table_def.go [hunk 1]
```diff
@@ -1471,12 +1471,12 @@ var MetricTableMap = map[string]MetricTableDef{
 	"tikv_engine_max_get_duration": {
 		PromQL:  `max(tikv_engine_get_micro_seconds{$LABEL_CONDITIONS}) by (db,type,instance)`,
 		Labels:  []string{"instance", "type", "db"},
-		Comment: "The max time consumed when executing get operations",
+		Comment: "The max time consumed when executing get operations, unit is microsecond",
```
Comment (lonng) [inline infoschema/metric_table_def.go]:
```suggestion
		Comment: "The max time consumed when executing get operations, in milliseconds",
```

Comment (lonng) [reply infoschema/metric_table_def.go]: Apply to other diffs.

Comment (crazycs520) [reply infoschema/metric_table_def.go]: great, done.

File: executor/inspection_result.go [hunk 2]
```diff
@@ -481,3 +494,155 @@ func (thresholdCheckInspection) inspect(ctx context.Context, sctx sessionctx.Con
...
+			sql = fmt.Sprintf("select instance, max(value) as max_value from metric_schema.%s %s group by instance having max_value > %f;", rule.tbl, condition, rule.threshold)
+		}
+		rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithContext(ctx, sql)
+		if err != nil {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("execute '%s' failed: %v", sql, err))
+			continue
+		}
+		for _, row := range rows {
+			actual := fmt.Sprintf("%.2f", row.GetFloat64(1))
+			expect := ""
```
Comment (Deardrops) [inline executor/inspection_result.go]:
```suggestion
			expected := ""
```

Comment (crazycs520) [reply executor/inspection_result.go]: done.

**Pull request merged**